internal/k8sdeps/funstruct.go

```diff
@@ -0,0 +1,114 @@
+package k8sdeps
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/kustomize/pkg/gvk"
+	"sigs.k8s.io/kustomize/pkg/ifc"
+	"strings"
+)
+
+var _ ifc.FunStruct = &KustFunStruct{}
+
+// KustFunStruct wraps Unstructed to isolate apimachinery dependence.
+type KustFunStruct struct {
+	unstructured.Unstructured
+}
+
+// GetGvk returns the Gvk name of the object.
+func (fs *KustFunStruct) GetGvk() gvk.Gvk {
+	return gvk.FromSchemaGvk(fs.GroupVersionKind())
+}
+
+// Copy provides a copy behind an interface.
+func (fs *KustFunStruct) Copy() ifc.FunStruct {
+	return &KustFunStruct{*fs.DeepCopy()}
+}
+
+// NewKustFunStructFromObject returns a new instance of KustFunStruct.
+func NewKustFunStructFromObject(obj runtime.Object) (*KustFunStruct, error) {
+	// Convert obj to a byte stream, then convert that to JSON (Unstructured).
+	marshaled, err := json.Marshal(obj)
+	if err != nil {
+		return nil, err
+	}
+	var u unstructured.Unstructured
+	err = u.UnmarshalJSON(marshaled)
+	// creationTimestamp always 'null', remove it
+	u.SetCreationTimestamp(metav1.Time{})
+	return &KustFunStruct{Unstructured: u}, err
+}
+
+// NewKustFunStructFromMap returns a new instance of KustFunStruct.
+func NewKustFunStructFromMap(m map[string]interface{}) *KustFunStruct {
+	return NewKustFunStructFromUnstruct(unstructured.Unstructured{Object: m})
+}
+
+// NewKustFunStructFromUnstruct returns a new instance of KustFunStruct.
+func NewKustFunStructFromUnstruct(u unstructured.Unstructured) *KustFunStruct {
```

Comment:
These `NewKustFunStructFrom...` functions are not part of the interface. Do you plan to add them to the interface or to a different interface? So that you can remove the dependency on `k8sdeps` from `resource`.